pkg/api/transport.go

```diff
@@ -11,15 +13,36 @@ import (
 // NewTransport initializes a new http.Transport.
 func NewTransport() *http.Transport {
 	return &http.Transport{
-		Proxy:               nil,
-		TLSHandshakeTimeout: DefaultTimeoutSecs * time.Second,
+		ForceAttemptHTTP2:   true,
+		MaxConnsPerHost:     1,
 		MaxIdleConns:        1,
 		MaxIdleConnsPerHost: 1,
-		MaxConnsPerHost:     1,
-		ForceAttemptHTTP2:   true,
+		Proxy:               nil,
+		TLSHandshakeTimeout: DefaultTimeoutSecs * time.Second,
 	}
 }
 
+// NewTransportWithCloudfareDNS initializes a new http.Transport with cloudfare DNS resolver.
+func NewTransportWithCloudfareDNS() *http.Transport {
+	t := NewTransport()
+
+	t.DialTLSContext = func(ctx context.Context, network, addr string) (net.Conn, error) {
+		r := &net.Resolver{
+			PreferGo: false,
+			Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+				d := net.Dialer{
+					Timeout: DefaultTimeoutSecs * time.Second,
+				}
+				return d.DialContext(ctx, network, "1.1.1.1:53")
```

Comment:
We can't have multiple resolvers right? Asking because some people only have ipv6 networks and the Cloudflare ipv4 address will never work for them.